src/Microsoft.AspNetCore.SignalR.Client.TS/Formatters.ts

```diff
@@ -0,0 +1,146 @@
+﻿import { Message, MessageType } from './Message';
+
+let knownTypes = {
+    "T": MessageType.Text,
+    "B": MessageType.Binary,
+    "C": MessageType.Close,
+    "E": MessageType.Error
+};
+
+function splitAt(input: string, searchString: string, position: number): [string, number] {
+    let index = input.indexOf(searchString, position);
+    if (index < 0) {
+        return [input.substr(position), input.length];
+    }
+    let left = input.substring(position, index);
+    return [left, index + searchString.length];
+}
+
+export namespace SSE {
+    export function parseMessage(input: string): [string, Message] {
```

Comment:
Ok, I can do that. I'm not a fan of using exceptions for parsing errors since they aren't necessarily exceptional cases and it's cleaner to do control flow with return values. I can change these around though.